src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCounterVisitor.java

```diff
@@ -0,0 +1,55 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class CommandCounterVisitor implements CommandVisitor {
+    
+    private Map<String, Integer> allCommandsCounter = new HashMap<>();
+
+    
+
+    @Override
+    public void visit(SetPositionCommand command) {
+       
+        incrementCommandsCounter(command.getClass().getSimpleName());
+    }
+    
+    public void incrementCommandsCounter(String command) {
+        
+    	allCommandsCounter.merge(command, 1, Integer::sum);
+    }
+
+    @Override
+    public void visit(OperateToCommand command) {
+        
+        incrementCommandsCounter(command.getClass().getSimpleName());
```

Comment:
OK, but this approach is not stateless. Visitor is not reusable, running visitation again on a another command would not reset the result.